### PR TITLE
Let Flake8 check examples and fix warnings

### LINKED ISSUE #948: Let flake8 check examples
``flake8`` would check for whitespace and lines >80 (both not caught by ``black`` in examples); currently ``flake8`` is only used for pygmt, setup.py, and doc/conf.py.

_Originally posted by @meghanrjones in https://github.com/GenericMappingTools/pygmt/issues/935#issuecomment-782408392_

Do we want flake8 to check all examples? Or do we want to limit line length in examples?

### PR DESCRIPTION
**Pull request by meghanrjones**
**Description of proposed changes**

This PR enables flake8 checking of the examples directory and fixes resultant warnings.

The diff is very large because of the line length configuration (opinions welcome for whether this is worth the benefit of having style checks for the examples).

Fixes #948 

**Reminders**

- [ ] Run `make format` and `make check` to make sure the code follows the style guide.
- [ ] Add tests for new features or tests that would have caught the bug that you're fixing.
- [ ] Add new public functions/methods/classes to `doc/api/index.rst`.
- [ ] Write detailed docstrings for all functions/methods.
- [ ] If adding new functionality, add an example to docstrings or tutorials.

**Slash Commands**

You can write slash commands (`/command`) in the first line of a comment to perform
specific operations. Supported slash commands are:

- `/format`: automatically format and lint the code
- `/test-gmt-dev`: run full tests on the latest GMT development version

### CODE REVIEW

**Review by seisman — APPROVED** [review]
I'm OK with these changes.

File: examples/gallery/symbols/datetime_inputs.py [hunk 1]
```diff
@@ -7,8 +7,9 @@
 - :class:`numpy.datetime64`
 - :class:`pandas.DatetimeIndex`
 - :class:`xarray.DataArray`: datetimes included in a *xarray.DataArray*
-- raw datetime strings in `ISO format <https://en.wikipedia.org/wiki/ISO_8601>`__
-  (e.g. ``"YYYY-MM-DD"``, ``"YYYY-MM-DDTHH"``, and ``"YYYY-MM-DDTHH:MM:SS"``)
+- raw datetime strings in
+  `ISO format <https://en.wikipedia.org/wiki/ISO_8601>`__ (e.g.
```
Comment (weiji14) [inline examples/gallery/symbols/datetime_inputs.py:11]:
```suggestion
  `ISO 8601 format <https://en.wikipedia.org/wiki/ISO_8601>`__ (e.g.
```

**Review by weiji14 — APPROVED** [review]
My eyes glossed over after scanning through diffs on 30-ish files... But ok with merging this big diff since you've done the hard work already. Thanks!

**Comment by meghanrjones** [comment]
Any ideas what's wrong with the windows tests?

**Comment by weiji14** [comment]
> Any ideas what's wrong with the windows tests?

The Windows tests fail randomly sometimes. Maybe restart the tests (usually I would just ignore it, but since this is a big PR, best be safe).

**Comment by seisman** [comment]
FYI, I just restarted the tests.

**Pull request merged**